packages/host/tests/integration/components/card-basics-test.gts

```diff
@@ -2100,6 +2101,64 @@ module('Integration | card-basics', function (hooks) {
       assert.dom('[data-test-customer-billAmount]').containsText('100');
     });
 
+    test('renders a card with polymorphic primitive field', async function (assert) {
+      class SpecialStringAEmbedded extends Component<typeof SpecialStringA> {
+        <template>
+          <span data-test-polymorphic='special-string-a'>{{@model}}</span>
+        </template>
+      }
+      class SpecialStringA extends StringField {
+        static displayName = 'SpecialStringA';
+        static embedded = SpecialStringAEmbedded;
+      }
+      class SpecialStringBEmbedded extends Component<typeof SpecialStringB> {
+        <template>
+          <span data-test-polymorphic='special-string-b'>{{@model}}</span>
+        </template>
+      }
+      class SpecialStringB extends StringField {
+        static displayName = 'SpecialStringB';
+        static embedded = SpecialStringBEmbedded;
+      }
+      class TestCard extends CardDef {
+        static displayName = 'TestCard';
+        @field specialField = contains(StringField);
+        static isolated = class Isolated extends Component<typeof TestCard> {
+          setSubclass = () => {
+            this.args.model[fields] = { specialField: SpecialStringB };
+            this.args.model.specialField = 'New Name';
+          };
+          <template>
+            <button {{on 'click' this.setSubclass}} data-test-set-subclass>Set
+              Subclass From Outside</button>
+            <@fields.specialField />
+          </template>
+        };
+      }
+
+      loader.shimModule(`${testRealmURL}test-cards`, {
+        SpecialStringA,
+        SpecialStringB,
+        TestCard,
+      });
+
+      let card = new TestCard({
+        specialField: 'Old Name',
+        [fields]: {
+          specialField: SpecialStringA,
+        },
+      });
```

Comment:
This is how you could initialize a polymorphic primitive field from the card's constructor.